core/node.go

```diff
@@ -253,6 +388,5 @@ func (n *Node) receive(ctx context.Context) (*Message, error) {
 // Close closes the Node and any active connections.
 func (n *Node) Close() error {
 	n.cancel()
-	n.sub.Cancel()
```

Comment:
We're passing in `n.ctx` to `n.sub` during initialization so it now closes automatically when we call `n.cancel`.